lib/gui/app.js

```diff
@@ -123,10 +125,10 @@ app.run((AnalyticsService, ErrorService, UpdateNotifierService, SelectionStateMo
 
     AnalyticsService.logEvent('Notifying update');
 
-    return UpdateNotifierService.notify();
-
+    return s3.getLatestVersion(s3.BUCKET_URL_PRODUCTION, {
```

Comment:
> Remember that the first S3 call is cached

Oh, I didn't realise that.

> I'm happy to refactor if this helps to clarify things.

Yeah, I think my suggestion makes the code more readable (without relying on the "magic" hidden caching) and I can't see any downsides?